### PR TITLE
i#7670 client TLS: Use private TLS for static client threads

### LINKED ISSUE #7670: aarch64 dr_client_thread crashes with static DR due to TLS being NULL
If a client thread is created on aarch64 it crashes immediately in tls_thread_init().
The code there does not handle static DR.

As part of addressing this we'll want to fix up the api.static_sideline test as that's our static DR client thread test: #297 covers fixing issues there.

### PR DESCRIPTION
**Pull request by derekbruening**
Use privload_tls_init() to allocate and initialize TLS for client threads with static DR, where otherwise the TLS would just be all zeroes. This allows running limited app code in client threads, which is useful in static DR deployments.

Added direct TLS and app code to the static_sideline test.
Also tested on internal setups where the all-zero prior setup is not enough to run some useful app code in large apps.

Fixes #7670

### CODE REVIEW

File: core/unix/os.c [hunk 1]
```diff
@@ -4231,8 +4243,24 @@ dr_create_client_thread(void (*func)(void *param), void *arg)
...
+    void *our_tls = (void *)read_thread_register(LIB_SEG_TLS);
+    if (INTERNAL_OPTION(private_loader)) {
+        write_thread_register(NULL);
+    } else {
+        /* Try to leverage privload TLS setup to ensure the client thread has
+         * some TLS in place in case it invokes some app code (though client-safe
+         * code is better, in some cases it is a big advantage to be able to run
+         * some limited app code in a static-DR setup).
+         */
+        void *child_tls = privload_tls_init(our_tls);
```
Comment (abhinav92003) [inline core/unix/os.c:4255]: Can we add some test coverage via suite/tests/api/static_sideline.c?

Comment (derekbruening) [reply core/unix/os.c:4255]: Done: I added both direct TLS access and some app code. (Took a while b/c the compiler optimized out my TLS at first; and in some testing I was on a 64K-page-size machine and realized the start of the app TLS copy wasn't quite right for such machines so fixed that.)

Comment (derekbruening) [reply core/unix/os.c:4255]: But now the new test fails on Ub20 (passes on 22) -- yet more debugging...this TLS work has not been easy to get it working on all these platforms.
